From b7fb2b85547118d7dda33fd51772c816ce5c0dc7 Mon Sep 17 00:00:00 2001
From: karim_adli <karimri@protonmail.com>
Date: Fri, 20 Feb 2026 22:21:33 +0300
Subject: [PATCH 5/8] wineboot: Add WINEBOOT_HIDE_DIALOG environment variable

---
 programs/wineboot/wineboot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/programs/wineboot/wineboot.c b/programs/wineboot/wineboot.c
index aa91d01..730025d 100644
--- a/programs/wineboot/wineboot.c
+++ b/programs/wineboot/wineboot.c
@@ -1531,7 +1531,9 @@ static HWND show_wait_window(void)
 {
     HWND hwnd = CreateDialogParamW( GetModuleHandleW(0), MAKEINTRESOURCEW(IDD_WAITDLG), 0,
                                     wait_dlgproc, (LPARAM)prettyprint_configdir() );
-    ShowWindow( hwnd, SW_SHOWNORMAL );
+    WCHAR *hide = _wgetenv( L"WINEBOOT_HIDE_DIALOG" );
+    if (hide == NULL)
+    	ShowWindow( hwnd, SW_SHOWNORMAL );
     return hwnd;
 }
 
-- 
2.54.0

